js/router.js

```diff
@@ -1,15 +1,153 @@
+import $ from 'jquery';
 import { Router } from 'backbone';
+import { getGuid } from './utils';
+import { getPageContainer } from './utils/selectors';
+import UserPage from './views/UserPage';
+import TransactionsPage from './views/TransactionsPage';
+import TemplateOnly from './views/TemplateOnly';
 
 export default class ObRouter extends Router {
-  constructor() {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    // Temporarily requiring a dummy 'users' collection. Eventually, we'll
+    // be getting this data via fetches to the server
+    if (!options.usersCl) {
+      throw new Error('Need me some dummy users!');
+    }
+
+    this.usersCl = options.usersCl;
+
+    if (!options.pageNavVw) {
+      throw new Error('Please provide the app\'s pageNavVw.');
+    }
+
+    this.pageNavVw = options.pageNavVw;
+
     const routes = [
-      // ['home/:state(/:searchText)', 'home'],
-      // ['userPage/:userID(/:state)(/:itemHash)(/:skipNSFWmodal)', 'userPage'],
-      // [/^@([^\/]+)(.*)$/, 'userPageViaHandle'],
-      // ['transactions/:state(/:orderID)(/:tabState)', 'transactions'],
-      // ['settings/:state', 'settings'],
+      [/^@([^\/]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'userViaHandle'],
+      [/^(Qm[a-zA-Z0-9]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'user'],
+      ['transactions', 'transactions'],
+      ['transactions/:tab', 'transactions'],
+      ['*path', 'pageNotFound'],
     ];
 
-    routes.forEach((route) => this.route.apply(this, route));
+    routes.slice(0)
+      .reverse()
```

Comment:
Because when we add them in this way using `route.apply`, they end up going in in reverse order. So, I'm just reversing the array so they go in the order we expect, which is how they're listed in code. Otherwise, our default pageNotFound route would be matched and wouldn't allow legitimate routes above it to be matched.